core/trino-spi/src/main/java/io/trino/spi/ptf/TableArgument.java

```diff
@@ -102,6 +102,71 @@ public boolean isPassThroughColumns()
         return passThroughColumns;
     }
 
+    public static Builder tableArgumentBuilder()
+    {
+        return new Builder();
+    }
+
+    public static final class Builder
+    {
+        private Optional<QualifiedName> name;
+        private RowType rowType;
+        private List<String> partitionBy = List.of();
+        private List<SortItem> orderBy = List.of();
+        private boolean rowSemantics;
+        private boolean pruneWhenEmpty;
+        private boolean passThroughColumns;
+
+        private Builder() {}
+
+        public Builder name(Optional<QualifiedName> name)
+        {
+            this.name = name;
+            return this;
+        }
+
+        public Builder rowType(RowType rowType)
+        {
+            this.rowType = rowType;
+            return this;
+        }
+
+        public Builder partitionBy(List<String> partitionBy)
+        {
+            this.partitionBy = partitionBy;
+            return this;
+        }
+
+        public Builder orderBy(List<SortItem> orderBy)
+        {
+            this.orderBy = orderBy;
+            return this;
+        }
+
+        public Builder rowSemantics()
+        {
+            this.rowSemantics = true;
+            return this;
+        }
+
+        public Builder pruneWhenEmpty()
+        {
+            this.pruneWhenEmpty = true;
+            return this;
+        }
+
+        public Builder passThroughColumns()
+        {
+            this.passThroughColumns = true;
+            return this;
+        }
```

Comment:
I prefer passing explicit value via parameter. That make use of builder easier if you want to set a flag field to a computed value (where you do not know at compile time if you want it to be true or false).

E.g.
```
  boolean selectedRowSemantics = determineIfWeShouldUseRowSemantics(...).
  TableArgument.builder()
       ...
       .rowSemantics(selectedRowSemantics)
       .build()
```